esxi/guest-create.go

```diff
@@ -198,8 +199,26 @@ func guestCREATE(c *Config, guest_name string, disk_store string,
 			net_param = " --network='" + virtual_networks[0][0] + "'"
 		}
 
-		ovf_cmd := fmt.Sprintf("ovftool --acceptAllEulas --noSSLVerify --X:useMacNaming=false "+
-			"-dm=%s --name='%s' --overwrite -ds='%s' %s '%s' '%s'", boot_disk_type, guest_name, disk_store, net_param, src_path, dst_path)
+		extra_params := "--X:injectOvfEnv --allowExtraConfig "
+		if (strings.HasSuffix(src_path, ".ova") || strings.HasSuffix(src_path, ".ovf")) {
+			// in order to process any OVF params, guest should be immediately powered on
+			// This is because the ESXi host doesn't have a cache to store the OVF parameters, like the vCenter Server does.
+			// Therefore, you MUST use the ‘--X:injectOvfEnv’ debug option with the ‘--poweron’ option
+			extra_params = extra_params + "--powerOn "
```

Comment:
Per discussion in  https://github.com/josenk/terraform-provider-esxi/issues/76,
--powerOn might be passed only if `len(ovf_properties) > 0`, i.e. at least one custom ovf_property was provided.

And ovf_properties is pretty often used features of ova images anyway.